src/drive/web/modules/viewer/NoViewerButton.jsx

```diff
@@ -56,9 +56,26 @@ const OpenWithCordovaButton = connect(
   />
 ))
 
-const DownloadButton = ({ t, file }, { client }) => (
+// WARNING: this is already duplicated in cozy-client and navigation/duck/action
+// Eventually, we would call a cozy-client method, so we duplicate the code for now to avoid ugly import
+const forceFileDownload = (href, filename) => {
+  const element = document.createElement('a')
+  element.setAttribute('href', href)
+  element.setAttribute('download', filename)
+  element.style.display = 'none'
+  document.body.appendChild(element)
+  element.click()
+  document.body.removeChild(element)
+}
+
+const downloadFile = async file => {
+  const downloadURL = await createDecryptedFileURL(file)
```

Comment:
I think here, we should check if the file is a regular file or if it is an encrypted one : otherwise we'll try to decipher any files.

Reply:
Well, yes, obviously...